packages/karma/karma_web_test.bzl

```diff
@@ -191,9 +191,13 @@ def run_karma_web_test(ctx):
     """
     files = depset(ctx.files.srcs)
     for d in ctx.attr.deps + ctx.attr.runtime_deps:
-        if hasattr(d, "node_sources"):
+        has_node_sources = hasattr(d, "node_sources")
+        has_dev_scripts = hasattr(d, "dev_scripts")
+        if has_node_sources:
             files = depset(transitive = [files, d.node_sources])
-        elif hasattr(d, "files"):
+        if has_dev_scripts:
+            files = depset(transitive = [files, d.dev_scripts])
+        if not has_node_sources and not has_dev_scripts and hasattr(d, "files"):
```

Comment:
this case is meant for a situation like
```
ts_web_test_suite(
    name = "test",
    srcs = [
        "@npm//node_modules/tslib:tslib.js",
    ],
    runtime_deps = [
        ":initialize_testbed",
    ],
    # do not sort
    bootstrap = [
        "@npm//node_modules/zone.js:dist/zone-testing-bundle.js",
        "@npm//node_modules/reflect-metadata:Reflect.js",
    ],
    browsers = [
        "@io_bazel_rules_webtesting//browsers:chromium-local",
    ],
    deps = [
        ":rxjs_umd_modules",
        ":test_lib",
        "@npm//karma-jasmine", <--- this dep
    ],
)
```
where the dep is a node module, in which case the original implementation would add all files in the package.
Here nothing gets added because the dep does have `node_sources`, albeit an empty one.

Reply:
Looks like the `karma-jasmine` dep is not even needed in the first place.